### ISSUE TITLE
"Safe to Send to" does not work when using Lava in Email Form block "From" field

### ISSUE THREAD

**Issue by JimMichael** [SparkDevNetwork/Rock#3124]
### Description
The _Safe to Send to_ option introduced in v7 does not seem to work.

### Steps to Reproduce
1. Go to Admin | Communications | Safe Sender Domains
1. Edit your Safe Sender domain to enable the _Safe to Send to_ option.
1. Use an email form block that delivers to someone@safesenderdomain.com, and make the from address ~an external (e.g. someone@gmail.com) address~ `{{ FormFields.Email }}` then use the form and enter a non-safe-sender from address.

**Expected behavior:**

Mail would arrive FROM the someone@gmail.com address, not from the substituted organization email address.

**Actual behavior:**

Mail arrives from the org email substituted address just as it does when you don't have _Safe to Send to_ enabled, and is identical to pre-v7 behavior. My understanding is that it will not substitute the from address IF you have this feature enabled and IF all recipients belong to the Safe Sender Domain, but my tests show this does not happen.

Note: We use the SendGrid email transport, in case that factors into this... maybe the feature involved a change to the Mailgun transport that never made it into the SendGrid plugin?

### Versions
* **Rock Version:** 7.4, 8.0

**Comment by mikejed** [comment]
This issue, too, only comes up when using this specific block AND when using {{ FormFields.Email }} in the block settings' "From" input. If you hard-code an external address in the settings, or use the normal communicate block, the "Safe to Send" setting works as advertised.

**Comment by JimMichael** [comment]
Note: This issue fixes itself when Lava is enabled on the FromEmail field per https://github.com/SparkDevNetwork/Rock/issues/3126

**Comment by JimMichael** [comment]
I've honestly lost track of this issue but from my testing it still seems to be an issue with Lava in the From/From Name fields of this block as of 8.4. As far as I can tell the current block still doesn't allow Lava, and we have to change these lines
                ```message.FromEmail = GetAttributeValue( "FromEmail" ).ResolveMergeFields( mergeFields, GetAttributeValue( "EnabledLavaCommands" ) );
                message.FromName = GetAttributeValue( "FromName" ).ResolveMergeFields( mergeFields, GetAttributeValue( "EnabledLavaCommands" ) );``` to add the "EnabledLavaCommands" stuff on the end to get them to work with the {{FormFields.Email }} syntax (grabbing the email from the form itself) but even then, trying to grab the CurrentPerson email or name like this still doesn't work
![image](https://user-images.githubusercontent.com/3903408/47308706-fb9fea80-d5f7-11e8-9b79-1d585fe87fe5.png)
and that's been broken since 7.0. For now we've had to resort to hidden FormFields where we grab the current person with Lava, and reference THAT (formfield) in the From address/name of on the block... I guess I'm trying to determine if allowing the normal CurrentPerson syntax will ever return?

**Comment by JimMichael** [comment]
This is still an issue as of 8.6alpha. Any chance it can get some attention? As it is, out of the box you cannot use the Email Form block to send an email and have it be "from" the sender (instead of being substituted).

To summarize, there are two issues:
1. Rock ships the EmailForm block without the ability to use Lava in the From Email and From Name fields. This is easily fixed by editing the .cs file to add the Lava ability on those fields (though if that is the "correct" fix, I can't say... I'm not a dev ;-)

2. Even when you DO manually adjust the file to enable Lava on those fields per above, Lava that _used to work just fine_ like `{{ CurrentPerson.Email }}` no longer functions within those fields. Instead, you have to do a cheesy work-around where you stuff the CurrentPerson.Email into a form field, THEN reference it via `{{ FormFields.Email }}`. I stress that this used to work fine until something broke it in v7, I think.

**Comment by JimMichael** [comment]
Doh, nevermind. I didn't realize I was testing on 8.4 and this has been fixed in 8.5 via #3126
